WordPress/gutenberg
# Background image: ensure consistency with defaults and fix reset/remove functionality

Follow up to:

- https://github.com/WordPress/gutenberg/pull/64192


## What and how?

Block background styles will be part of WordPress 6.7.

This PR fixes miscellaneous bugs in the background image default values and controls. It:

- ensures that inherited values are shown in the controls
- ensures that any default values for **uploaded images** are represented in the controls
- ~~blocks should honour any inherited styles. That means, defaults should not overwrite styles defined in theme.json or global styles~~
- changes the default values of "center" to "50% 50%" so that it is represented in the focus picker control values. 
- ensures "resetting" an block's styles resets any current styles (at the moment it does not)
- ensures "removing" an image resets any current styles and remove access to the background controls



## Why?

To address a lot of inconsistencies in the way defaults are applied to block background styles.


## Testing Instructions

This might take some time, sorry!

Using a theme with background images set on blocks (see example theme.json), add some blocks to the editor (see example blocks).

Ensure that you can see inherited styles in the controls and that no background default values are applied.

Now upload/add your own images to these blocks.

1. The background size value should be set to "cover"
2. When toggling between background sizes, the defaults should be visible, E.g., Switch to "contain" and the image should be centered ("50% 50%"), and the value reflected in the controls.
3. Save and reset the image. Make sure that all values on the image have been reset.
4. Change the background size or other value for the inherited image. 
5. Now, "remove" the image so that the background is none. All properties, expect "background-image:none" should be gone.

Ensure that the frontend matches what you're seeing in the editor.

Site-wide background images should operate as before and be unaffected by any defaults aside from the existing position of `50% 0` position when switching to tiled.


<details>

<summary>Example block HTML</summary>
```html

<blockquote class="wp-block-quote">
<p class="has-text-color has-link-color" style="color:#f8f8f8;font-size:51px">Quote</p>
</blockquote>



<pre class="wp-block-verse has-white-color has-text-color has-link-color" style="font-size:61px">Verse</pre>



<div class="wp-block-group">
<p class="has-text-color has-link-color" style="color:#fefefe;font-size:59px">Group</p>
</div>

```
</details>

<details>

<summary>example theme.json</summary>

```json

{
	"$schema": "../../schemas/json/theme.json",
	"version": 3,
	"settings": {
		"appearanceTools": true,
		"layout": {
			"contentSize": "840px",
			"wideSize": "1100px"
		}
	},
	"styles": {

		"background": {
			"backgroundImage": {
				"url": "https://images.pexels.com/photos/105819/pexels-photo-105819.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr="
			}
		},
		"blocks": {
			"core/verse": {
				"background": {
					"backgroundImage": {
						"url": "https://images.pexels.com/photos/27332202/pexels-photo-27332202/free-photo-of-surfer.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2"
					},
					"backgroundSize": "contain"
				},
				"dimensions": {
					"minHeight": "100px"
				}
			},
			"core/quote": {
				"background": {
					"backgroundImage": {
						"url": "https://images.pexels.com/photos/22484288/pexels-photo-22484288/free-photo-of-the-circular-stone-terraces-of-the-inca-ruins.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2"
					}
				},
				"dimensions": {
					"minHeight": "100px"
				}
			},
			"core/group": {
				"background": {
					"backgroundImage": {
						"url": "https://images.pexels.com/photos/27101553/pexels-photo-27101553/free-photo-of-a-cocktail-with-mint-and-lime-in-it.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2"
					}
				},
				"dimensions": {
					"minHeight": "111px"
				}
			}
		}
	}
}

```

</details>


### Testing Instructions for Keyboard


## Screenshots or screencast 



https://github.com/user-attachments/assets/8e04f760-9811-42e7-8a2a-dca698aa73d9




## TODO

- [ ] The UI is getting complicated. We probably need E2E tests in a follow up.
